### PR TITLE
phonebook: Persist initial phonebook peers; remove unused ExtendPeerList

### PR DESCRIPTION
**Pull request by Eric-Warehime**
## Summary
We never actually call ExtendPeerList anywhere. When we refresh the SRV list periodically it just fully replaces the peer list.

## Test Plan

grep'ing ExtendPeerList shows nothing. Also existing tests should pass.

### CODE REVIEW

**Comment by algorandskiy** [comment]
If I remember right, it does not replace but merges. Otherwise `phonebook.json` and `-p` would not work?

**Comment by Eric-Warehime** [comment]
> If I remember right, it does not replace but merges. Otherwise `phonebook.json` and `-p` would not work?

All occurrences in the repo according to github search:
https://github.com/search?q=repo%3Aalgorand%2Fgo-algorand%20ExtendPeerList&type=code

SRV updates use this to replace peer list on refresh.
https://github.com/algorand/go-algorand/blob/master/network/wsNetwork.go#L1775-L1785C2

Replace peer list on ws network create--which is where the -p overrides get passed in.
https://github.com/algorand/go-algorand/blob/master/network/wsNetwork.go#L2452C12-L2452C27

**Comment by algorandskiy** [comment]
This sounds like a buggy behavior for me. `-p` entries must survive and not being replaced.

**Comment by iansuvak** [comment]
I agree that we probably want clear semantics on the `-p` and the method naming.This change itself is safe but as @algorandskiy said we want to make sure that DNS refresh never clobbers manual peers provided via `-p`. Might be outside of the scope of this cleanup PR though.

**Comment by algorandskiy** [comment]
>Might be outside of the scope of this cleanup PR though.

We'll need Extend function for this...

**Review by winder — APPROVED** [review]
Seems fine.

File: network/wsNetwork.go [hunk 1]
```diff
@@ -1775,12 +1775,12 @@ func (wn *WebsocketNetwork) refreshRelayArchivePhonebookAddresses() {
 func (wn *WebsocketNetwork) updatePhonebookAddresses(relayAddrs []string, archiveAddrs []string) {
 	if len(relayAddrs) > 0 {
 		wn.log.Debugf("got %d relay dns addrs, %#v", len(relayAddrs), relayAddrs[:imin(5, len(relayAddrs))])
-		wn.phonebook.ReplacePeerList(relayAddrs, string(wn.NetworkID), PhoneBookEntryRelayRole)
+		wn.phonebook.ExtendPeerList(relayAddrs, string(wn.NetworkID), PhoneBookEntryRelayRole)
```
[the remark this answers is not in the document]
Comment (iansuvak) [reply network/wsNetwork.go:1778]: That does sound ok to me. If a relay get's removed from SRV it's either bad or unreachable, if it's unreachable it isn't doing real harm and if it's bad it's still only one of 4 relays (unless we remove multiple at once)

**Review by algorandskiy — APPROVED** [review]

File: network/wsNetwork.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (algorandskiy) [reply network/wsNetwork.go:1778]: yeah, not great, then maybe we need a new "seeded" entries concept that survive in `ReplacePeerList` ?

**Review by iansuvak — DISMISSED** [review]

**Review by winder — APPROVED** [review]

**Review by algorandskiy — CHANGES_REQUESTED** [review]
Blocking this, looks like folks think it should use replace logic with preserving seeded entries.

**Comment by Eric-Warehime** [comment]
> Blocking this, looks like folks think it should use replace logic with preserving seeded entries.

That sounds like it should be done in a separate PR, but if we can't merge this without it then I can add it here.

**Comment by algorandskiy** [comment]
> That sounds like it should be done in a separate PR, but if we can't merge this without it then I can add it here.

If we merge as is then dead relays could appear be in phonebook, not end of the world but still. This PR is just 5 LoC and the new fix will touch the same code, so I suggest to fix it right here.

**Review by algorandskiy — APPROVED** [review]

**Review by iansuvak — APPROVED** [review]
LGTM -- I like the property based testing

**Pull request merged**